dot/parachain/collator-protocol/validator_side.go

```diff
@@ -397,6 +778,43 @@ func (f fetchedCollationInfo) String() string {
 		f.relayParent.String(), f.paraID, f.candidateHash.Value.String(), f.collatorID)
 }
 
+func (f *fetchedCollationInfo) FromString(str string) error {
+	splits := strings.Split(str, ",")
+	if len(splits) != 4 {
+		return fmt.Errorf("invalid string format for fetched collation info")
```

Comment:
create a sentinel error and append the wrong string in the error context
```suggestion
		return fmt.Errorf("%w: %s", ErrInvalidStringFormat, str)
```

Reply:
creating sentinel error. appending the string is just too much information. Anyways it's the same string in the argument.